### DISCUSSION TITLE
ELF CrakeMe 解题讨论

### DISCUSSION THREAD

**Discussion by wumingzhilian** [HUSTSeclab/software-security-dojo#20] [General]
我们在这里讨论 ELF CrakeMe 模块中遇到的问题

**Comment by RutingZhang0429** [comment]
询问一下level1.0的elf header编写问题
除了将elf-crackme-level1.0改为elf64和exec类型以外，还需要怎样修改

**Reply by ZAIQuQ** [reply]
前面用pwntools的时候`context(arch="amd64", os="linux", log_level='debug')`就已经暗示machine是`amdx86-64`，其实exec不需要改，只需要改 Magic里面的elf64 和 Machine 。

**Reply by RutingZhang0429** [reply]
好的谢谢，我尝试的时候改了elf64，exec和machine，没想到exec不用改

**Comment by wyywwi** [comment]
level2.0 可以用 level2.1 的办法解决，虽然会报 hash 不对，但是可以执行 elf-crackme-level2.0 出结果

**Reply by lowa9** [reply]
所以正确(或者说期望的)的解法是啥嘞

**Reply by RutingZhang0429** [reply]
通过修改.data段的内容获取flag

**Comment by RutingZhang0429** [comment]
level2.2使用暴力破解时，中途进程自动终止，怎么解决自动终止的问题
<img width="352" alt="image" src="https://github.com/HUSTSeclab/software-security-dojo/assets/129151845/cb661af2-339c-486d-8504-70b348898ac5">

**Reply by RutingZhang0429** [reply]
想到的一种方法是修改代码，手动从终止处继续运行

**Reply by ZAIQuQ** [reply]
我碰到过类似的情况，把log_level由debug改成info，或者把log信息输入到txt里面，防止终端被log信息挤满。

**Reply by wyywwi** [reply]
我把utility.py和elf的二进制文件都下载到了本地，修改utility的代码来尝试在本地爆破运行，这样可以规避无用的log并提升爆破速度，这样也可以解决

**Comment by zllancc** [comment]
level3.0修改了plt跳转地址的，elf-crackme-level3.0能正常运行但exec会报错
![图片](https://github.com/HUSTSeclab/software-security-dojo/assets/150664801/ca9d9c16-c1ea-4bae-92cc-dac963c15d2e)
![图片](https://github.com/HUSTSeclab/software-security-dojo/assets/150664801/48f0a299-937b-40b5-8e85-c9ed5bbbe7a9)
不知道是哪里出错了

**Reply by ZAIQuQ** [reply]
这个是环境的问题，就是execve函数调用cat的时候找不到它了，把路径转到/usr/bin可以解决

**Reply by zllancc** [reply]
解决了，谢谢

**Reply by wyywwi** [reply]
谢谢，刚刚也碰到了这个问题，还以为是链接的地方push的参数要改，琢磨半天🤣
